src/main/java/duke/Ui.java

```diff
@@ -37,40 +38,45 @@ public void close() throws IOException {
         bf.close();
     }
 
-    public void greet() {
+    public String greet() {
         System.out.println(GREETING);
+        return "Hello from Duke! Nice to see you";
     }
 
-    public void goodBye() {
+    public String goodBye() {
         lineUI();
         System.out.println(BYE);
         lineUI();
+
+        return BYE;
     }
 
-    public void listCommand() {
+    public String listCommand() {
         ArrayList<Task> arr = TaskList.getList();
```

Comment:
When using ArrayList, it might be better to use plural form for the variable name. Besides, arr might not be a very meaning variable name, might consider change it.